dev/release/00-prepare.sh

```diff
@@ -121,6 +121,16 @@ update_versions() {
       parquet/README.md
   rm -f parquet/README.md.bak
   git add parquet/README.md
+
+  # Update version number for datafusion README
+  sed -i.bak -E -e \
```

Comment:
@kou any idea how to test the release procedure? The release scripts are starting to grow pretty huge, so We should have an automatized way to test it.

Reply:
How about the followings?

1. Create a new branch from `origin/master`
2. Run `dev/release/00-prepare.sh`
3. Check the output of `git diff origin/master..apache-arrow-${new_version}`
    (It must have changes for the new version without "-SNAPSHOT")
4. Check the output of `git diff apache-arrow-${new_version}..`
    (It must have changes for the next version with "-SNAPSHOT")